src/ios-deploy/ios-deploy.m

```diff
@@ -1051,12 +1053,12 @@ void launch_debugger(AMDeviceRef device, CFURLRef url) {
         signal(SIGHUP, SIG_DFL);
         signal(SIGLLDB, SIG_DFL);
         child = getpid();
-
+        pid_t oldfgpid = 0;
         int pfd[2] = {-1, -1};
-        if (isatty(STDIN_FILENO))
+        if (interactive && isatty(STDIN_FILENO))
```

Comment:
It seems like we'd still want to call bring_process_to_foreground() in non-interactive mode. We'd want to send SIGTTOU.

Have you tried this in non-interactive mode?

Reply:
Sorry for late answer. 

For me it works. lldb never tried to configure terminal. But I agree, if it can lead to problems, it should be omited.  I'll update the Pull request.

Reply:
Thanks, this looks good